pkg/build/generator/generator.go

```diff
@@ -513,6 +513,11 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 				Kind: "DockerImage",
 				Name: runtimeImageName,
 			}
+			// We're using single secret for pulling builder and runtime images.
+			// If builder image doesn't need a secret, let's check whether runtime image needs it or not.
+			if build.Spec.Strategy.SourceStrategy.PullSecret == nil {
+				build.Spec.Strategy.SourceStrategy.PullSecret = g.resolveImageSecret(ctx, builderSecrets, build.Spec.Strategy.SourceStrategy.RuntimeImage, bc.Namespace)
+			}
```

Comment:
i'd suggest calling it twice, once for each image.  if you extend it to take 2 image specs, then we'll have a case where we want to check for 3 image specs, etc.